lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,360 @@
+      child: SingleChildScrollView(
+        reverse: true,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => _SelectedUserChip(userId: userId)),
+              IntrinsicWidth(child: _buildSearchField(context)),
+            ])),
+        ])));
+  }
+}
+
+class _SelectedUserChip extends StatelessWidget {
+  const _SelectedUserChip({required this.userId});
+
+  final int userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final user = store.getUser(userId);
+
+    return Container(
+      decoration: BoxDecoration(
+        color: designVariables.bgMenuButtonSelected,
+        borderRadius: BorderRadius.circular(3)),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Avatar(userId: userId, size: 22, borderRadius: 3),
+          Padding(
+            padding: const EdgeInsetsDirectional.fromSTEB(5, 3, 4, 3),
+            child: Text(user?.fullName ?? zulipLocalizations.unknownUserName,
+              style: TextStyle(
+                fontSize: 16,
+                height: 16 / 16,
+                color: designVariables.labelMenuButton))),
+        ]));
+  }
+}
+
+class _NewDmUserList extends StatelessWidget {
+  const _NewDmUserList({
+    required this.filteredUsers,
+    required this.selectedUserIds,
+    required this.onUserTapped,
+  });
+
+  final List<User> filteredUsers;
+  final Set<int> selectedUserIds;
+  final void Function(int userId) onUserTapped;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    if (filteredUsers.isEmpty) {
+      return Center(
+        child: Padding(
+          padding: const EdgeInsets.all(24),
+          child: Text(
+            zulipLocalizations.newDmSheetNoUsersFound,
+            style: TextStyle(
+              color: designVariables.labelMenuButton,
+              fontSize: 16))));
+    }
+
+    return ListView.builder(
+      padding: EdgeInsets.all(8),
+      itemCount: filteredUsers.length,
+      itemBuilder: (context, index) {
+        final user = filteredUsers[index];
+        final isSelected = selectedUserIds.contains(user.userId);
+
+        return InkWell(
+          splashFactory: NoSplash.splashFactory,
+          onTap: () => onUserTapped(user.userId),
+          child: Container(
+            decoration: isSelected ? BoxDecoration(
+              color: designVariables.bgMenuButtonSelected,
+              borderRadius: BorderRadius.circular(10)) : null,
```

Comment:
nit: `Container` shouldn't be necessary; we can use `DecoratedBox`

another nit: I think we usually have the `null` case first, for reasons similar to why early returns are preferred.